src/aws-cpp-sdk-core/include/aws/core/client/AWSClientAsyncCRTP.h

```diff
@@ -40,6 +42,77 @@ namespace Client
     class ClientWithAsyncTemplateMethods
     {
     public:
+        ClientWithAsyncTemplateMethods()
+         : m_isInitialized(true),
+           m_operationsProcessed(0)
+        {
+            AwsServiceClientT* pThis = static_cast<AwsServiceClientT*>(this);
```

Comment:
could we use [shared from this](https://en.cppreference.com/w/cpp/memory/enable_shared_from_this) instead of a raw ptr?